drivers/place/template_mailer.cr

```diff
@@ -0,0 +1,243 @@
+require "placeos-driver"
+require "placeos-driver/interface/mailer"
+require "placeos-driver/interface/mailer_templates"
+
+# This driver uses metadata templates to send emails via the SMTP mailer.
+# It should be configured as Mailer_1 with the next mailer in the chain as Mailer_2.
+# 
+# It also updates metadata in the staff API with available fields for use in email templates.
+class Place::TemplateMailer < PlaceOS::Driver
+  include PlaceOS::Driver::Interface::Mailer
+  include PlaceOS::Driver::Interface::MailerTemplates
+
+  descriptive_name "Template Mailer"
+  generic_name :Mailer
+  description %(uses metadata templates to send emails via the SMTP mailer)
+
+  default_settings({
+    cache_timeout: 300, # timeout for the template cache
+    timezone:    "GMT",
+    update_schedule:    "*/20 * * * *", # cron schedule for updating template fields
+  })
```

Comment:
Would it make sense to add an option to only update fields, so not delete fields from drivers that are no longer available.
Or make them sticky for x number of updates before being removed, to avoid any intermittent issues.

Reply:
added an option to make the fields sticky for x number of updates